include/dos_system.h

```diff
@@ -343,10 +343,11 @@ class DOS_Drive {
 	DOS_Drive();
 	virtual ~DOS_Drive() = default;
 
-	virtual bool FileOpen(DOS_File** file, const char* name, uint8_t flags) = 0;
-	virtual bool FileCreate(DOS_File** file, const char* name,
-	                        FatAttributeFlags attributes) = 0;
-	virtual bool FileUnlink(const char* _name)=0;
+	virtual std::unique_ptr<DOS_File> FileOpen(const char* name,
+	                                           uint8_t flags) = 0;
+	virtual std::unique_ptr<DOS_File> FileCreate(const char* name,
+	                                             FatAttributeFlags attributes) = 0;
```

Comment:
`const  FatAttributeFlags attributes` ?

Reply:
Also gets modified inside some of the functions.

Reply:
Also, not sure why we care if this argument is const.  It's pass by value so it doesn't matter to the caller.

Reply:
As explained above. It's not from the point of the view of the caller.

It's from the point of view from the _function_. You'd make a local var const when applicable, even it's an int or a bool, no? The same reasoning goes for the incoming function args, otherwise they become "free to mutate for all" vars for no good reason 😄 

But, I get it it's can be hassle to refactor old code because you need to introduce temporaries then inside the function. I've been doing that, personally, but you don't have to do it now if you don't feel like it. But ultimately we should for consistency at some point IMO.